app/workers/fast_track/disability_compensation_job.rb

```diff
@@ -52,16 +51,47 @@ def perform(form526_submission_id, full_name)
       end
     end
 
+    # class AccountNotFoundError < RuntimeError; end
+
     private
 
-    def bp_readings(client)
-      @bp_readings ||= client.list_resource('observations')
-      @bp_readings.present? ? FastTrack::HypertensionObservationData.new(@bp_readings).transform : []
-    end
+    class LighthouseVetsHealth
+      def initialize(form526_submission)
+        @form526_submission = form526_submission
+      end
+
+      def client
+        @client ||= Lighthouse::VeteransHealth::Client.new(account.icn.presence)
+      end
+
+      def account
+        user_uuid = @form526_submission.user_uuid.presence
+        edipi = @form526_submission.auth_headers['va_eauth_dodedipnid'].presence
+        account = Account.find_by(idme_uuid: user_uuid) if user_uuid
+        account ||= Account.find_by(logingov_uuid: user_uuid) if user_uuid
+        account ||= Account.find_by(edipi: edipi) if edipi
+        # Should we raise and error instead of expecting a NoMethodError?
+        # raise AccountNotFoundError.new("for user_uuid: #{user_uuid}") unless account&.icn
```

Comment:
Opinions on this question? See the RSpec that expect `NoMethodError`. If we know the cause for the error (i.e., `AccountNotFoundError`), we should make it clear in the error message.

Reply:
Yeah, I think that would be great - we could perhaps change that in a separate PR and track the change over in our Notion dashboard. Sound good?

Reply:
Second what @julialeague said here 👏  It'd be great to get an improved test in.